### ISSUE TITLE
Spinner in button with vertical margin

### ISSUE THREAD

**Issue by pts-eduardoacuna** [the-unsullied/react-fancy-button#4]
###  Description

A `FancyButton` with a margin on top can mess up the centering of the spinner.

```jsx
<FancyButton
    style={{ marginTop: '24px!important' }}
    type='submit'
    disabled={...}
    label={...}
    onClick={...}
/>
```

The spinner get's centered taking into account the height and margin of the button:

<img width="468" alt="screen shot 2017-07-25 at 3 28 53 pm copy" src="https://user-images.githubusercontent.com/28717018/28596808-808aa8d4-714f-11e7-8154-69f08856d0fc.png">

**Comment by pts-moog16** [comment]
Yes we should allow the dev to pass styles through the props, just like we do with classes, since inline styles are popular. 

Please open PR and I will review.
